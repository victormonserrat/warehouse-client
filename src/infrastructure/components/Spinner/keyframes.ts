import { keyframes } from 'styled-components'

export const dash = keyframes`
    0% {
        stroke-dasharray: 0, 9.25rem;
        stroke-dashoffset: 0;
    }
    50% {
        stroke-dasharray: 5.5rem, 9.25rem;
        stroke-dashoffset: -2.25rem;
    }
    100% {
        stroke-dasharray: 5.5rem, 9.25rem;
        stroke-dashoffset: -7.75rem;
    }
`

export const rotate = keyframes`
    100% { 
        transform: rotate(360deg);
    }
`
