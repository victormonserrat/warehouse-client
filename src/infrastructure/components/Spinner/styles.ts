import styled from 'styled-components'

import { dash, rotate } from './keyframes'

export const Component = styled.svg`
  animation: ${rotate} 1.5s linear infinite;
`

export const Circle = styled.circle`
  animation: ${dash} 1.5s ease-in-out infinite;
  fill: none;
  stroke: ${({ theme }) => theme.colors.blue};
  stroke-linecap: round;
  stroke-width: 0.3125rem;
`
