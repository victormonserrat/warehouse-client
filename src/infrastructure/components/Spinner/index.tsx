import { memo, SVGAttributes } from 'react'

import { Circle, Component } from './styles'

const Spinner = ({ ...props }: SVGAttributes<SVGSVGElement>) => (
  <Component aria-label="Loading" viewBox="0 0 50 50" {...props}>
    <Circle cx="25" cy="25" r="20"></Circle>
  </Component>
)

export default memo(Spinner)
