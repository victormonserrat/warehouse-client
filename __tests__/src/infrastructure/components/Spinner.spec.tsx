import Spinner from '~/infrastructure/components/Spinner'
import { render } from '~/tests/utils'

describe('Spinner', () => {
  it('renders correctly', () => {
    const { container } = render(<Spinner />)

    expect(container).toMatchSnapshot('Regular')
  })
})
