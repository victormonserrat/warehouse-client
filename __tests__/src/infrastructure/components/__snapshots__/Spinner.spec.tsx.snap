// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly: Regular 1`] = `
<div>
  <svg
    aria-label="Loading"
    class="sc-bczRLJ hswLeu"
    viewBox="0 0 50 50"
  >
    <circle
      class="sc-gsnTZi fByHmj"
      cx="25"
      cy="25"
      r="20"
    />
  </svg>
</div>
`;
